projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -184,11 +190,16 @@ export class IgxOverlayService implements OnDestroy {
             this.setupModalWrapper(info);
         }
 
-        if (info.settings.positionStrategy.settings.openAnimation) {
-            this.playOpenAnimation(info);
-        } else {
-            this.onOpened.emit({ id: info.id, componentRef: info.componentRef });
-        }
+        //  show overlay in set time out. This allow us to return the id before overlay appears
+        setTimeout(() => {
```

Comment:
Slight issue - `onOpening` still emits way in advance, so you can't actually intercept it with the returned id.

Reply:
done